pkg/cmd/server/api/latest/helpers.go

```diff
@@ -83,5 +85,25 @@ func ReadYAMLFile(filename string, obj runtime.Object) error {
 	if err != nil {
 		return err
 	}
-	return Codec.DecodeInto(data, obj)
+	err = Codec.DecodeInto(data, obj)
+	return captureSurroundingJSONForError(fmt.Sprintf("could not load config file %q due to a error: ", filename), data, err)
+}
+
+func captureSurroundingJSONForError(prefix string, data []byte, err error) error {
+	if syntaxErr, ok := err.(*json.SyntaxError); err != nil && ok {
```

Comment:
Impossible, encoding/json is a disaster.  Going to be a rip and replace someday with a real JSON parser.

----- Original Message -----
> > @@ -83,5 +85,25 @@ func ReadYAMLFile(filename string, obj runtime.Object)
> > error {
> >  	if err != nil {
> >  		return err
> >  	}
> > -	return Codec.DecodeInto(data, obj)
> > +	err = Codec.DecodeInto(data, obj)
> > +	return captureSurroundingJSONForError(fmt.Sprintf("could not load config
> > file %q due to a error: ", filename), data, err)
> > +}
> > +
> > +func captureSurroundingJSONForError(prefix string, data []byte, err error)
> > error {
> > +	if syntaxErr, ok := err.(*json.SyntaxError); err != nil && ok {
> 
> How hard is it to produce line numbers for the lazy?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2718/files#r31729785